util/src/main/java/com/netcracker/ConsoleHelper.java

```
@@ -0,0 +1,39 @@
+package com.netcracker;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+public class ConsoleHelper {
+    private static final BufferedReader READER = new BufferedReader(new InputStreamReader(System.in));
+
+    private ConsoleHelper(){
+        throw new RuntimeException("Don't create instance of this class");
```

That exception looks like an overkill to me ;)
Why would anyone want to create an instance of that class inside itself (private ensures that it is the only option left) ? Only if they make it a singleton instead of a static class, but then they know what they are doing, its very unlikely that they will mess it up.

---

Why not? Actually, it's from Effective Java, where Joshua Bloch write that utility classes should throw RuntimeException, so no one can't create instance of this class.

---

Well, it drains reader's attention, compared to empty private constructor.

---

Yeah, this is the main reason, why should we throw an exception - to say user that he shouldn't create instance of this class in any case (By reflection in this case) [article from book](http://www.informit.com/articles/article.aspx?p=1216151&seqNum=4)

---

&#x1F34F; Revoked